src/hooks/loadables/useLoadSpendingLimits.ts

```diff
@@ -76,11 +75,11 @@ export const useLoadSpendingLimits = (): AsyncResult<SpendingLimitState[]> => {
   const chainId = useChainId()
   const provider = useWeb3ReadOnly()
 
-  // Update spending limits whenever a transaction is executed
+  // Update spending limits whenever the tx history updates
   // TODO: Find a more optimised way to update them
   useEffect(() => {
-    return txSubscribe(TxEvent.SUCCESS, () => setUpdateSpendingLimitsTag(Date.now()))
-  }, [])
+    setUpdateSpendingLimitsTag(Date.now())
+  }, [safe.txHistoryTag])
```

Comment:
You don't need a state/effect for this. Just make `safe.txHistoryTag` itself a dependency of the update hook.

Reply:
Nice!